src/share.tsx

```diff
@@ -17,18 +19,116 @@ function Header(props: PropsWithChildren): JSX.Element {
+function isDID(value: string): boolean {
+  try {
+    DID.parse(value.trim())
+    return true
+  } catch {
+    return false
+  }
+}
+
+function isEmail(value: string): boolean {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+  return !isDID(value) && emailRegex.test(value)
+}
+
+export function ShareSpace({ spaceDID }: { spaceDID: SpaceDID }): JSX.Element {
   const [{ client }] = useW3()
   const [value, setValue] = useState('')
   const [downloadUrl, setDownloadUrl] = useState('')
+  const [sharedEmails, setSharedEmails] = useState<{ email: string, capabilities: string[] }[]>([])
+
+  const updateSharedEmails = (delegations: { email: string, capabilities: string[] }[]) => {
+    setSharedEmails(prev => {
+      const newEmails = delegations.filter(d => !prev.some(item => item.email === d.email))
+      return [...prev, ...newEmails]
+    })
+  }
+
+  useEffect(() => {
+    if (client && spaceDID) {
+      // Find all delegations via email where the spaceDID is present
+      const delegations = client.delegations()
+        .filter(d => d.capabilities.some(c => c.with === spaceDID))
+        .filter(d => d.audience.did().startsWith('did:mailto:'))
+        .map(d => ({
+          email: DIDMailTo.toEmail(DIDMailTo.fromString(d.audience.did())),
+          capabilities: d.capabilities.map(c => c.can)
+        }))
+      updateSharedEmails(delegations)
+    }
+  }, [client, spaceDID])
+
+  async function shareViaEmail(email: string): Promise<void> {
+    if (!client) {
+      throw new Error(`Client not found`)
+    }
+
+    const currentSpace = client.agent.currentSpace()
+    try {
+      const space = client.spaces().find(s => s.did() === spaceDID)
+      if (!space) {
+        throw new Error(`Could not find space to share`)
+      }
+
+      const delegatedEmail = DIDMailTo.email(email)
+
+      // FIXME (fforbeck): enable shareSpace function call after @w3ui/react lib is updated to v2.4.0 and the issue with blobs are solved
+      // const delegation = await client.shareSpace(delegatedEmail, space.did())
+
+      // Make sure the agent is using the shared space before delegating
+      await client.agent.setCurrentSpace(spaceDID)
+
+      // Delegate capabilities to the delegate account to access the **current space**
+      const delegation = await client.createDelegation(
+        {
+          did: () => DIDMailTo.fromEmail(delegatedEmail),
+        },
+        [
+          'space/*',
+          'store/*',
+          'upload/*',
+          'access/*',
+          'usage/*',
+          'filecoin/*',
+        ],
+      )
+
+      const sharingResult = await client.capability.access.delegate({
+        space: spaceDID,
+        delegations: [delegation],
+      })
+
+      if (sharingResult.error) {
+        throw new Error(
+          `failed to share space with ${delegatedEmail}: ${sharingResult.error.message}`,
+          {
+            cause: sharingResult.error,
+          }
+        )
+      }
```

Comment:
@alanshaw I had to implement the logic here until we solve that blob issue.

Reply:
Gah...sorry :(